Assets/Scripts/Buddy/BuddyStats.cs

```diff
@@ -83,20 +137,40 @@ public void GiveResource( PlayerStats actorStats, ResourceData resourceData)
 	{
 		DebugUtils.Assert( _isAlive, "Cannot give a dead buddy resources." );
 
+		if ( resourceData == null )
```

Comment:
`resourceData` should never be null by the time it gets to this point so this should be an assert.